SharedModules/mpi_func.f90

```diff
@@ -0,0 +1,198 @@
+module mpi_func
+  use numPrecision
+#ifdef MPI
+  use mpi_f08
+#endif
+  use errors_mod,        only : fatalError
+
+  implicit none
+
+  integer(shortInt), private   :: worldSize = 1
+  integer(shortInt), private   :: rank = 0
+  integer(shortInt), parameter :: MASTER_RANK = 0
+
+  !! Public type that replicates exactly particleState
+  !!
+  !! It is necessary for load balancing in the dungeon: particles have to be
+  !! transferred betwen processes, and MPI doesn't allow to transfer types with
+  !! type-bound procedures
+  type, public :: particleStateDummy
```

Comment:
This is so painful :( Are there no alternatives around this? It feels like a nasty bug waiting to happen!

Reply:
I don't think so. We need to pass particle states between ranks. The alternative is to pass individual properties but I don't see how that's less painful.